cmd/dagger/flags.go

```diff
@@ -629,115 +642,138 @@ func (v *platformValue) Get(ctx context.Context, dag *dagger.Client, _ *dagger.M
 
 // AddFlag adds a flag appropriate for the argument type. Should return a
 // pointer to the value.
-func (r *modFunctionArg) AddFlag(flags *pflag.FlagSet) (any, error) {
+func (r *modFunctionArg) AddFlag(flags *pflag.FlagSet) error {
 	name := r.FlagName()
 	usage := r.Description
 
 	if flags.Lookup(name) != nil {
-		return nil, fmt.Errorf("flag already exists: %s", name)
+		return fmt.Errorf("flag already exists: %s", name)
 	}
 
 	switch r.TypeDef.Kind {
 	case dagger.StringKind:
 		val, _ := getDefaultValue[string](r)
-		return flags.String(name, val, usage), nil
+		flags.String(name, val, usage)
+		return nil
 
 	case dagger.IntegerKind:
 		val, _ := getDefaultValue[int](r)
-		return flags.Int(name, val, usage), nil
+		flags.Int(name, val, usage)
+		return nil
 
 	case dagger.BooleanKind:
 		val, _ := getDefaultValue[bool](r)
-		return flags.Bool(name, val, usage), nil
+		flags.Bool(name, val, usage)
+		return nil
 
 	case dagger.ScalarKind:
 		scalarName := r.TypeDef.AsScalar.Name
 
 		if val := GetCustomFlagValue(scalarName); val != nil {
 			flags.Var(val, name, usage)
-			return val, nil
+			return nil
 		}
 
 		val, _ := getDefaultValue[string](r)
-		return flags.String(name, val, usage), nil
+		flags.String(name, val, usage)
+		return nil
 
 	case dagger.ObjectKind:
 		objName := r.TypeDef.AsObject.Name
 
 		if val := GetCustomFlagValue(objName); val != nil {
 			flags.Var(val, name, usage)
-			return val, nil
+			return nil
 		}
 
 		// TODO: default to JSON?
-		return nil, fmt.Errorf("unsupported object type %q for flag: %s", objName, name)
+		return &UnsupportedFlagError{
+			Name: name,
+			Type: fmt.Sprintf("%q object", objName),
+		}
```

Comment:
Does the new return type address the `TODO: default to JSON?`

Reply:
No, feature wise it's the same. I just replaced a generic error with a custom one in order to tell the difference later.